include/seqan3/search/algorithm/search.hpp

```diff
@@ -15,10 +15,91 @@
 #include <seqan3/core/algorithm/configuration.hpp>
 #include <seqan3/range/views/persist.hpp>
 #include <seqan3/search/algorithm/detail/search.hpp>
+#include <seqan3/search/algorithm/detail/search_traits.hpp>
 #include <seqan3/search/fm_index/all.hpp>
 #include <seqan3/std/algorithm>
 #include <seqan3/std/ranges>
 
+
+namespace seqan3::detail
+{
+
+/*!\brief Class used to validate the search configuration.
+ * \ingroup submodule_search_algorithm
+ */
+struct search_configuration_validator
+{
+    /*!\brief Validates the error configuration.
+     *
+     * \tparam configuration_t The type of the search configuration.
+     *
+     * \param[in] cfg The configuration to validate.
+     *
+     * \throws std::invalid_argument
+     *
+     * \details
+     *
+     * Checks if the given thresholds for the configured errors are valid. Otherwise throws std::invalid_argument.
+     */
+    template <typename configuration_t>
+    static void validate_error_configuration(configuration_t const & cfg)
```

Comment:
I wrote a card on this topic, FYI: https://github.com/seqan/seqan3/projects/6#card-31194436